src/compile/mark/mixins.ts

```diff
@@ -314,42 +314,39 @@ export function bandPosition(
 ) {
   const scaleName = model.scaleName(channel);
   const sizeChannel = channel === 'x' ? 'width' : 'height';
+  const {markDef, encoding, config} = model;
 
-  if (model.encoding.size != null || model.markDef.size != null || defaultSizeRef?.value !== undefined) {
-    const orient = model.markDef.orient;
-    if (orient) {
-      const centeredBandPositionMixins = {
-        // Use xc/yc and place the mark at the middle of the band
-        // This way we never have to deal with size's condition for x/y position.
-        [channel + 'c']: ref.fieldRef(fieldDef, scaleName, {}, {band: 0.5})
-      };
+  const vgChannel = alignedChannel(channel, markDef, config);
 
-      if (getTypedFieldDef(model.encoding.size)) {
-        return {
-          ...centeredBandPositionMixins,
-          ...nonPosition('size', model, {vgChannel: sizeChannel})
-        };
-      } else if (isValueDef(model.encoding.size)) {
+  const centeredBandPositionMixins = {
+    [vgChannel]: ref.fieldRef(fieldDef, scaleName, {}, {band: 0.5})
+  };
+
+  if (encoding.size != null || markDef.size != null) {
+    const orient = markDef.orient;
+    if (orient) {
+      if (getTypedFieldDef(encoding.size) || isValueDef(encoding.size)) {
         return {
           ...centeredBandPositionMixins,
           ...nonPosition('size', model, {vgChannel: sizeChannel})
         };
-      } else if (model.markDef.size !== undefined) {
+      } else if (markDef.size !== undefined) {
         return {
           ...centeredBandPositionMixins,
-          [sizeChannel]: {value: model.markDef.size}
-        };
-      } else if (defaultSizeRef && defaultSizeRef.value !== undefined) {
-        return {
-          ...centeredBandPositionMixins,
-          [sizeChannel]: defaultSizeRef
+          [sizeChannel]: {value: markDef.size}
         };
       }
     } else {
-      log.warn(log.message.cannotApplySizeToNonOrientedMark(model.markDef.type));
+      log.warn(log.message.cannotApplySizeToNonOrientedMark(markDef.type));
     }
   }
 
+  if (defaultSizeRef && defaultSizeRef?.value !== undefined) {
```

Comment:
I think `if(defaultSizeRef?.value !== undefined)` is sufficient here

Reply:
Right, I added ? so I can remove the first part, but then forgot to remove lol